### PR TITLE
Refactor SriovNetwork and SriovIBNetwork controllers to use same code

### PR DESCRIPTION
**Pull request by ykulazhenkov**
Controllers were refactoring to use genericNetworkReconciler, which contains all logic

This is necessary to make it easier to add new controllers.

### CODE REVIEW

**Comment by ykulazhenkov** [comment]
used by #646

File: controllers/generic_network_controller.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2021.
```
Comment (e0ne) [inline controllers/generic_network_controller.go:2]: nit: please change to 2024 if you do another update

**Review by e0ne — APPROVED** [review]
Good cleanup, thanks @ykulazhenkov

**Comment by ykulazhenkov** [comment]
@zeeke @SchSeba 
Could you please review this PR? :)
Thx!

**Pull request merged**